es-app/src/VirtualSystemManager.cpp

```diff
@@ -0,0 +1,415 @@
+{
+	std::vector<std::string> themeSys = getSystemsFromTheme();
+	return std::find(themeSys.begin(), themeSys.end(), folder) != themeSys.end();
+}
+
+std::vector<std::string> VirtualSystemManager::getUnusedSystemsFromTheme()
+{
+	std::vector<std::string> cfgSys = getSystemsFromConfig();
+	std::vector<std::string> themeSys = getSystemsFromTheme();
+	for(auto sysIt = themeSys.begin(); sysIt != themeSys.end(); )
+	{
+		if (std::find(cfgSys.begin(), cfgSys.end(), *sysIt) != cfgSys.end())
+		{
+			sysIt = themeSys.erase(sysIt);
+		}
+		else
+		{
+			sysIt++;
+		}
+	}
+	return themeSys;
+}
+
+FileData::SortType VirtualSystemManager::getSortType(std::string desc) {
+	std::vector<FileData::SortType> SortTypes = FileSorts::SortTypes;
+	// find it
+	for(unsigned int i = 0; i < FileSorts::SortTypes.size(); i++)
+	{
+		const FileData::SortType& sort = FileSorts::SortTypes.at(i);
+		if(sort.description == desc)
+		{
+			return sort;
+		}
+	}
+	// if not found default to name, ascending
+	return FileSorts::SortTypes.at(0);
+}
+
+std::vector<std::string> VirtualSystemManager::stringToSystems(std::string systemString)
+{
+	// from a comma separated string, get a vector of system names
+	std::vector<std::string> strs;
+	boost::split(strs, systemString, boost::is_any_of(","));
+	return strs;
+}
+
+void VirtualSystemManager::loadVirtualSystems()
+{
+	loadAutoVirtualSystems();
+	// we will also load custom systems here in the future
+	loadCustomVirtualSystems();
+	// Now see which ones are enabled
+	loadEnabledListFromSettings();
+	// add to the main System Vector, and create Views as needed
+	updateSystemsList();
+}
+
+void VirtualSystemManager::updateSystemsList()
+{
+	// remove all Virtual Systems
+	for(auto sysIt = SystemData::sSystemVector.begin(); sysIt != SystemData::sSystemVector.end(); )
+	{
+		if ((*sysIt)->isVirtual())
+		{
+			sysIt = SystemData::sSystemVector.erase(sysIt);
+		}
+		else
+		{
+			sysIt++;
+		}
+	}
+
+	// add enabled ones
+	for(std::map<std::string, VirtualSystemData>::iterator it = mAllVirtualSystems.begin() ; it != mAllVirtualSystems.end() ; it++ )
+	{
+		if(it->second.isEnabled)
+		{
+			SystemData::sSystemVector.push_back(it->second.system);
+		}
+	}
+
+	// remove disabled gamelist views
+	// create gamelist views if needed
+	// iterate the map
+}
+
+void VirtualSystemManager::loadAutoVirtualSystems()
+{
+	for(std::map<std::string, VirtualSystemDecl>::iterator it = mVirtualSystemDecls.begin() ; it != mVirtualSystemDecls.end() ; it++ )
+	{
+		VirtualSystemDecl sysDecl = it->second;
+		if (!sysDecl.isCustom && !findVirtualSystem(sysDecl.name))
+		{
+			SystemData* newSys = new SystemData(sysDecl.name, sysDecl.longName, mVirtualEnvData, sysDecl.themeFolder, true);
+
+			FileData* rootFolder = newSys->getRootFolder();
+			FileFilterIndex* index = newSys->getIndex();
+			for(auto sysIt = SystemData::sSystemVector.begin(); sysIt != SystemData::sSystemVector.end(); sysIt++)
+			{
+				if ((*sysIt)->getName() != "retropie") {
```

Comment:
Sure. It may make it a tad slower (iterating a list and compare with each) but sounds fair.

I take it you mean, still, a centralized definition of the exclusions (but still in the code), rather than a user-selectable options-based systems-exclusion list?